Python/bytecodes.c

```diff
@@ -2218,23 +2218,21 @@ dummy_func(
             JUMPBY(1-oparg);
             #if ENABLE_SPECIALIZATION
             here[1].cache += (1 << OPTIMIZER_BITS_IN_COUNTER);
-            if (here[1].cache > tstate->interp->optimizer_backedge_threshold &&
-                // Double-check that the opcode isn't instrumented or something:
-                here->op.code == JUMP_BACKWARD &&
-                // _PyOptimizer_BackEdge is going to change frame->prev_instr,
-                // which breaks line event calculations:
-                next_instr->op.code != INSTRUMENTED_LINE
-                )
-            {
+            if (here[1].cache > tstate->interp->optimizer_backedge_threshold) {
+                assert(here->op.code == JUMP_BACKWARD);
                 OBJECT_STAT_INC(optimization_attempts);
-                frame = _PyOptimizer_BackEdge(frame, here, next_instr, stack_pointer);
-                if (frame == NULL) {
-                    frame = tstate->current_frame;
+                _Py_CODEUNIT *src = here;
+                while(oparg > 255) {
+                    oparg >>= 8;
+                    src--;
+                }
```

Comment:
I don't think this assert is correct, since it could be an `INSTRUMENTED_LINE` or something. Ditto for the matching assert in `_PyOptimizer_BackEdge`.

Reply:
Ouch. Can instrumentation really overwrite `EXTENDED_ARG`? @markshannon